packages/diagram/lib/diagram.dart

```diff
@@ -0,0 +1,442 @@
+  ///
+  /// The animation will start at timestamp [start], end at [end], with the time
+  /// between frames specified by [frameDuration].  For instance, to generate
+  /// 11 frames of an animation that is 1 second long, set [start] to
+  /// [Duration.zero] (the default), set [end] to `const Duration(seconds: 1)`,
+  /// and set [frameDuration] to `const Duration(milliseconds: 100)`. The
+  /// animation includes the frames at [start] and [end].
+  ///
+  /// None of the parameters may be null. The [start] parameter must be less
+  /// than or equal to the [end] parameter, and the [frameDuration] must be
+  /// less than or equal to the time between [start] and [end]. All durations
+  /// must be greater than [Duration.zero].
+  Future<List<File>> drawAnimatedDiagramToFiles({
+    Duration start: Duration.zero,
+    @required Duration end,
+    @required Duration frameDuration,
+    ui.ImageByteFormat format: ui.ImageByteFormat.png,
+  }) async {
+    assert(end != null);
+    assert(start != null);
+    assert(frameDuration != null);
+    assert(end >= start);
+    assert(frameDuration <= (end - start));
+    assert(frameDuration > Duration.zero);
+    assert(end > Duration.zero);
+    assert(start >= Duration.zero);
+
+    Duration now = start;
+    int index = 0;
+    List<File> outputFiles = <File>[];
+    while (now <= end) {
+      File outputFile = _getFrameFilename(now, index);
+      ui.Image captured = await drawDiagramToImage();
+      ByteData encoded = await captured.toByteData(format: format);
+      List<int> bytes = encoded.buffer.asUint8List().toList();
+      print('Writing frame $index ($now), ${bytes.length} bytes, ${captured.width}x${captured.height} '
+          '${_byteFormatToString(format)}, to ${outputFile.absolute.path}');
+      outputFile.writeAsBytesSync(bytes);
+      advanceTime(frameDuration);
+      outputFiles.add(outputFile);
+      now += frameDuration;
+      ++index;
+    }
+    return outputFiles;
+  }
+
+  static File _basicFrameFilenameGenerator(Duration timestamp, int index) {
+    return new File('frame_${index.toString().padLeft(3, '0')}.png');
+  }
+
+  String _byteFormatToString(ui.ImageByteFormat format) {
+    switch (format) {
+      case ui.ImageByteFormat.rawRgba:
+        return 'RAW RGBA';
+      case ui.ImageByteFormat.rawUnmodified:
+        return 'NATIVE';
+      case ui.ImageByteFormat.png:
+        return 'PNG';
+    }
+    return null;
+  }
+
+  File _getFrameFilename(Duration timestamp, int index) {
+    File outputFile = frameFilenameGenerator(timestamp, index);
+    if (!outputFile.isAbsolute && outputDirectory != null) {
+      // If output path is relative, make it relative to the output directory.
+      outputFile = new File(path.join(outputDirectory.absolute.path, outputFile.path));
+    }
+    assert(outputFile.path.endsWith('.png'), 'Since the only supported output format is PNG, filenames must end in ".png"');
```

Comment:
this seems to contradict the earlier comment?

Reply:
Yeah, stale comment. Fixed.